pkg/terra/client/gas_price_estimator.go

```diff
@@ -50,20 +50,17 @@ func (gpe *ClosureGasPriceEstimator) GasPrices() (map[string]sdk.DecCoin, error)
 var _ GasPricesEstimator = (*FCDGasPriceEstimator)(nil)
 
 type FCDGasPriceEstimator struct {
-	fcdURL url.URL
+	cfg    Config
 	client http.Client
 	lggr   Logger
 }
 
-func NewFCDGasPriceEstimator(fcdURLRaw string, requestTimeout time.Duration, lggr Logger) (*FCDGasPriceEstimator, error) {
-	// Sanity check the URL works and populate the cached value
-	fcdURL, err := url.Parse(fcdURLRaw)
-	if err != nil {
-		return nil, err
-	}
+type Config interface{ FCDURL() url.URL }
+
+func NewFCDGasPriceEstimator(cfg Config, requestTimeout time.Duration, lggr Logger) *FCDGasPriceEstimator {
```

Comment:
Why send full config in when we just use the URL? Makes users go through additional hoops to prepare the struct, which seem unnecessary.

Reply:
The config is dynamic, so this will now pick up runtime reconfiguration.

Reply:
Aha, the interface serves as a pointer to the latest config. Got it!

Reply:
Maybe additionally consider adding a relevant comment somewhere, so it's clear why we pass in config interfaces like this.